hnix-store-core/tests/NarFormat.hs

```diff
@@ -40,30 +56,40 @@ spec_narEncoding :: Spec
 spec_narEncoding = do
 
   -- For a Haskell embedded Nar, check that (decode . encode === id)
-  let roundTrip n = runGet getNar (runPut $ putNar n) `shouldBe` n
+  let
+    withTempDir act = Temp.withSystemTempDirectory "nar-test-XXXXXX" act
+
+    roundTrip :: String -> Nar -> IO ()
+    roundTrip narFileName n = withTempDir $ \tmpDir -> do
+        let narFilePath = tmpDir </> narFileName
+        unpackNarIO narEffectsIO (runPut (putNar n)) narFilePath
```

Comment:
This ignores the Either result type hiding parse errors in case of `sampleExecutable`, make sure to error with Left err message.

Not sure why it is happening yet
```haskell
ParseErrorBundle {
  bundleErrors = TrivialError 72 (Just (Tokens (10 :| [0,0,0,0,0,0,0,101,120,101,99,117,116,97,98]))) 
  (fromList [Tokens (8 :| [0,0,0,0,0,0,0,99,111,110,116,101,110,116,115])]) :| [],
  bundlePosState = PosState {pstateInput = "\r\NUL\NUL\NUL\NUL\NUL\NUL\NULnix-archive-1\NUL\NUL\NUL\SOH\NUL\NUL\NUL\NUL\NUL\NUL\NUL(\NUL\NUL\NUL\NUL\NUL\NUL\NUL\EOT\NUL\NUL\NUL\NUL\NUL\NUL\NULtype\NUL\NUL\NUL\NUL\a\NUL\NUL\NUL\NUL\NUL\NUL\NULregular\NUL\n\NUL\NUL\NUL\NUL\NUL\NUL\NULexecutable\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\b\NUL\NUL\NUL\NUL\NUL\NUL\NULcontents\"\NUL\NUL\NUL\NUL\NUL\NUL\NUL#!/bin/bash\n\ngcc -o hello hello.c\n\NUL\NUL\NUL\NUL\NUL\NUL\SOH\NUL\NUL\NUL\NUL\NUL\NUL\NUL)\NUL\NUL\NUL\NUL\NUL\NUL\NUL", 
  pstateOffset = 0, pstateSourcePos = SourcePos {sourceName = "", sourceLine = Pos 1, sourceColumn = Pos 1}, pstateTabWidth = Pos 8, pstateLinePrefix = ""}}
```